File: src/ubb/dp1920/examples/recap/polymorphism/Polymorphism.py
class Animal(object):
    def talk(self):
        print("?")

class Dog(Animal):
    def talk(self):
        print("woof woof!")

class Cat(Animal):
    def talk(self):
        print("meow!")

a = Animal()
c = Cat()
d = Dog()

a.talk()
c.talk()
d.talk()

# Both value AND type are transferred 
a = c
c.talk()

File: src/ubb/dp1920/examples/recap/polymorphism/Polymorphism.java
package ubb.dp1920.examples.recap.polymorphism;

class Animal {
    public void talk() {
        System.out.println("animal");
    }
};

class Cat extends Animal {
    public void talk() {
        System.out.println("meow");
    }
};

class Dog extends Animal {
    public void talk() {
        System.out.println("woof woof!");
    }
};

public class Polymorphism {
    public static void main(String[] args) {
        System.out.println("Java sample");
        // References to direct classes
        Animal a = new Animal();
        Cat c = new Cat();
        Dog d = new Dog();
        System.out.println("Pointers to class itself");
        a.talk();
        c.talk();
        d.talk();

        // References to the base class
        System.out.println("Pointers to base class");
        a = c;
        a.talk();
    }
}